NoaTeomi/Yoga
# Feature/docker setup

**Add Docker setup for the Yoga project**

Description:
This pull request introduces a Docker setup for the Yoga project, enabling the application to run in a containerized environment. Below are the key details of the setup:

**Key Changes:**

  -   **Dockerfile:** Created a Dockerfile to build a container for the Django project.    

  - Base image:  python:3.11-slim
  - Installs dependencies and sets environment variables.
   - Copies the project files, runs migrations, collects static files, and exposes port 8000.
   - Configured the Django server to run on 0.0.0.0:8000 within the container.

**Volume Mount:** 

Set up volume mounting for the Yoga/media/poses directory to ensure uploaded images and media files persist on the host machine.

**Project Structure:**

- Ensures static and media file handling works inside the container.
- The containerized setup does not affect local development.

**How to Test:**
1. Build the Docker image: 
     docker build -t yoga-sequencing-app .

2. Run the container:
      docker run -d -p 8000:8000 -v $(pwd)/Yoga/media/poses:/app/media/poses yoga-sequencing-app

3. Access the application at http://localhost:8000.

4. Ensure that:

-  The app runs properly inside the container.
-  Media files (poses) are displayed correctly.
-  Static files are served as expected.

**Additional Notes:**

- This setup allows the application to run both locally and inside Docker, with no disruption to local development workflows.
- Future enhancements could include Docker Compose integration and multi-stage builds for production.
